astro/tvm/versionrepo.go

```diff
@@ -132,14 +142,19 @@ func (r *VersionRepo) exists(version string) bool {
 }
 
 func (r *VersionRepo) getLock(version string) *sync.Mutex {
-	v, _ := r.downloadLock.LoadOrStore(version, &sync.Mutex{})
+	v, _ := r.lock.LoadOrStore(version, &sync.Mutex{})
 	return v.(*sync.Mutex)
 }
 
 // Get takes a version and returns the path to the Terraform binary for
 // that version. If the binary doesn't exist, it will be downloaded from
 // the Terraform website automatically.
 func (r *VersionRepo) Get(version string) (string, error) {
+	lock := r.getLock(version)
```

Comment:
The mutex is not there to provide safety to the map. The mutex is there to prevent us from downloading and installing the same version of Terraform several times concurrently.

If multiple threads request the same version of Terraform at the same time and it doesn't exist, only one will download it and the others will block until that is complete and then immediately return if the download is successful.

I've updated the comments to make this clearer. Can you let me know if this makes more sense now?